crates/bevy_ecs/src/observer/runner.rs

```diff
@@ -0,0 +1,226 @@
+use crate::{
+    component::{ComponentHooks, ComponentId, StorageType},
+    observer::{ObserverDescriptor, ObserverTrigger},
+    prelude::*,
+    query::DebugCheckedUnwrap,
+    system::{IntoObserverSystem, ObserverSystem},
+    world::DeferredWorld,
+};
+use bevy_ptr::PtrMut;
+
+/// Contains [`Observer`] information. This defines how a given observer behaves. It is the
+/// "source of truth" for a given observer entity's behavior.
+pub struct ObserverState {
+    pub(crate) descriptor: ObserverDescriptor,
+    pub(crate) runner: ObserverRunner,
+    pub(crate) last_trigger_id: u32,
+    pub(crate) despawned_watched_entities: u32,
+}
+
+impl Default for ObserverState {
+    fn default() -> Self {
+        Self {
+            runner: |_, _, _| {},
+            last_trigger_id: 0,
+            despawned_watched_entities: 0,
+            descriptor: Default::default(),
+        }
+    }
+}
+
+impl ObserverState {
+    /// Adds the given `event`
```

Comment:
The docs on all of these items need to give more information about what this actually does and why one might want to do it.